lib/core/elastic_connector_actions.rb

```diff
@@ -88,6 +88,38 @@ def claim_job(connector_id)
         job['_id']
       end
 
+      def find_current_job(connector_id)
```

Comment:
I like that it's calling the server for this, it's future-proof for multi-nodes environments.
One thing I wonder is about race conditions on the index.
you probably want to check the seq_num here to make sure no one updated the doc in the interim

https://www.elastic.co/guide/en/elasticsearch/reference/current/optimistic-concurrency-control.html

Reply:
In practice it means that when you start a new sync job, you pass along `_seq_no` from the old state you had, and if you get a rejection you know someone else started a new sync job already, and you back off

Reply:
Done.